Kajabi/sage-lib
# feat(button group): add wrap option

## Description

Adds optional `wrap` param to `SageButtonGroup`, allowing content to wrap to container widths.

- Sets *all* `SageButtonGroup` containers to `flex-wrap: wrap` (whether the param is enabled or not) to prevent mobile screen widths from horizontally scrolling. For breakpoints larger than `sm-min`, wrapping remains disabled by default
- Replaces use of `margin` with `gap` (as of Safari v14.1/14.7, `gap` is now supported in flex containers for all modern browsers) 
- `margin` has also been replaced with `gap` for `SageChoice` inside a `SageButtonGroup` 


## Screenshots

|  | Before  |  After  |
|---|-----|--------|
| sage example|![before](https://user-images.githubusercontent.com/816579/129403773-9d5f843c-c725-4312-911f-9d03833858a2.png)|![after](https://user-images.githubusercontent.com/816579/129403802-90b0128d-64a4-444c-9521-9cff405b888c.png)|
|in app|![before](https://user-images.githubusercontent.com/816579/129415929-ac125840-28fc-4e5e-b1f1-08a48b55fa78.png)|![after](https://user-images.githubusercontent.com/816579/129415944-82ed7af2-6fa0-49b3-a984-16c8ebf09d9f.png)|



## Testing in `sage-lib`
1. View the [Sage button example page](http://localhost:4000/pages/component/button) and scroll to the "Button group: Allow wrap" section
2. Click the "Toggle button group wrap" button to activate wrapping on all button group examples
3. Decrease the browser window width, and confirm that buttons wrap as expected
4. Click the "Toggle button group wrap" button once more to deactivate wrapping
5. Decrease the browser window width, and confirm that buttons extend past the confines of their container. Note that below the `sm-min` breakpoint width, wrapping is still enabled regardless of the setting (see note in description)


## Testing in `kajabi-products`

1. (**MED/HIGH**) Adds option for button groups to wrap to multiple lines. Default behavior on mobile views has been changed to allow wrapping (other breakpoints are unaffected), so existing uses should be checked to ensure content still flows as expected. Examples include:
   - [ ] Website Pages and Website Landing Pages: Edit page view - Publish status
   - [ ] Coupons: Delete coupon confirmation modal
   - [ ] Offers: Offer export modal
   - [ ] Email campaigns: Broadcast send type selection


## Related
- closes #690
